src/main/java/net/minecraftforge/client/model/animation/FastTESR.java

```diff
@@ -34,7 +34,7 @@
 public abstract class FastTESR<T extends TileEntity> extends TileEntitySpecialRenderer<T>
 {
     @Override
-    public final void render(T te, double x, double y, double z, float partialTicks, int destroyStage, float partial)
+    public void render(T te, double x, double y, double z, float partialTicks, int destroyStage, float partial)
```

Comment:
why allow overriding?

Reply:
read the third bullet point

Reply:
Now the question is, why would you want to extend FastTESR when writing a mixed TESR? The only thing in it as a render() dummy that isn't used unless you use your FastTESR as a normal TESR...

Reply:
The point of fast tesr is that this ISNT overrideable so that we can batch things, so no.

Reply:
In my case, I'd like to use an `AnimationTESR` for its fast rendering while adding additional normal stuff for the second pass. I know I *could* just delegate the fast render call to an `AnimationTESR` instance, but then I also have to take care that it is properly initialized (currently only the render dispatcher needs to be passed through, but that might change in the future).

I do not see the reason for restricting overriding the normal render method. The implementation in `FastTESR` just enables callers to ignore that the TESR was designed for batching, which IMO isn't the renderer's job. Allowing overriding the method will not mess with the batching in any way and improve code reusability.

As an aside, the implementation of `render` in `FastTESR` could even lead to people thinking their renderer was batched when it really wasn't, since they forgot to override `hasFastRenderer`. In that regard I'd personally also be in favour of either making the rendering a noop or even throwing an exception.

Reply:
BTW, I do like that there's code there that tells the modder what GL state they will have when their FastTESR is run. Otherwise it'd be a big guesswork, as the code that really sets the state is not easy to find. However, as the FastTESR is at the moment, it gives no clue at all that that render() method isn't actually executed, hiding away the real benefit and reason to use the FastTESR. And there's not even a hint of the fact you have to add a method to your TE, too.

(And btw/OT, what's with the disableCull()? Doesn't that cost rendering performance for the very rare case of free-standing quads?)